com.unity.mobile.notifications/Runtime/iOS/Plugins/UnityNotificationData.m

```diff
@@ -298,7 +298,7 @@ void _ReadNSDictionary(void* csDict, void* nsDict, void (*callback)(void* csDcit
 
 void _ReadAttachmentsNSArray(void* csList, void* nsArray, void (*callback)(void*, const char*, const char*))
 {
-    NSArray<UNNotificationAttachment*>* attachments = (__bridge_transfer NSArray<UNNotificationAttachment*>*)nsArray;
+    NSArray<UNNotificationAttachment*>* attachments = (__bridge NSArray<UNNotificationAttachment*>*)nsArray;
```

Comment:
This change is not correct, it causes a memory leak.
Is nsArray null when it crashes?

Reply:
nsArray is not null - this is being triggered twice for each notification (leading to a double free).

While the app is in the foreground we are listening for remote notifications (so that they don't trigger) - and also using local notifications.  This is causing UnityNotificationManager::userNotificationCenter:willPresentNotification to trigger both iOSNotificationsWrapper.NotificationReceived and iOSNotificationsWrapper.RemoteNotificationReceived causing this function to be executed twice with the same array.

I believe the proper solution is to make this change here and properly clean up the attachments in freeiOSNotificationData.